contractcourt/anchor_resolver.go

```diff
@@ -0,0 +1,135 @@
+package contractcourt
+
+import (
+	"errors"
+	"io"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/sweep"
+)
+
+// anchorResolver is a resolver that will attempt to sweep our anchor output.
```

Comment:
I think this may actually need to live independent of any given channel arb. Otherwise, they'll never transition from pending close to fully closed, since there'll be a lingering contract that's unresolved.

Reply:
Add select clause to mark as resolved when the sweeper gives up. By that time, the 16 block csv delay has long expired, so if fees ever come down far enough, the anchor will probably be swept by one of the cleaning services that we envision.

Reply:
When will the sweeper "give up"? Will check out the diff for more details...

Reply:
There is a comment about this in the `Resolve` method